### PR TITLE
Add Cloudflare bot authorization

### PR DESCRIPTION
**Pull request by ArtOfCode-**
In mitigating for the incident over the weekend, I made the server simply ignore if it couldn't fetch other Codidact sites and move on. This has left us without a functioning communities list in the footer. This change adds an authorization header that Cloudflare should recognise and allow the request.

### CODE REVIEW

**Comment by trichoplax** [comment]
I'm not sure if `communities.json` is used anywhere else, but a recent pull request removed the list of communities from the footer, so it isn't needed for that purpose. This was #1402 and `develop` has been deployed since that was merged, so production is no longer displaying communities in the footer.

**Comment by cellio** [comment]
The community switcher also pulls a list of communities from somewhere, presumably here (but I haven't checked).

**Comment by ArtOfCode-** [comment]
I can't say I know off the top of my head everywhere this is used, but seems sensible to restore it to working order until we can review properly whether it's still required.

**Comment by cellio** [comment]
Let's get this reviewed by someone more competent than I am. :-)  How do we test it?

File: config/initializers/zz_codidact_sites.rb [hunk 1]
```diff
@@ -4,7 +4,13 @@
   if Rails.env.development? || Rails.env.test?
     []
   else
-    response = Net::HTTP.get_response(URI('https://codidact.com/communities.json'))
+    uri = URI('https://codidact.com/communities.json')
+    req = Net::HTTP::Get.new(uri)
```
Comment (Oaphi) [inline config/initializers/zz_codidact_sites.rb:8]: Let's maybe take the opportunity and untie development from codidact being up by making this configurable and not hardcoded? Also makes sense if we ever switch domains or restructure. If I am able to get some time off for that, I can push here or barring that - separately.

**Review by Oaphi — APPROVED** [review]
LGTM

**Pull request merged**